examples/src/bin/gio_task.rs

```diff
@@ -0,0 +1,188 @@
+//! # GTask Sample
+//!
+//! This sample demonstrates how to implement a GIO async/finish API based on GTask (see
+//! https://developer.gnome.org/gio/stable/GTask.html) in Rust using the GTask generated bindings.
+//! This can be useful, for example, when porting to Rust some existing C code exposing such an API.
+
+use glib::subclass;
+use glib::subclass::prelude::*;
+use glib::translate::ToGlibPtr;
+use gtk::glib::translate::FromGlibPtrNone;
+use gtk::prelude::AsyncResultExt;
+use gtk::prelude::Cast;
+use gtk::prelude::FileExt;
+use gtk::{gio, glib};
+use std::convert::TryInto;
+
+pub mod imp {
+    use super::*;
+
+    // FileSize is a simple object that will just contain the read file size.
+    // Initially the size field will be initialized to 0.
+    pub type FileSize = <FileSizePrivate as ObjectSubclass>::Instance;
+    pub type FileSizeClass = <FileSizePrivate as ObjectSubclass>::Class;
+    pub struct FileSizePrivate {
+        pub size: std::cell::RefCell<i64>,
+    }
+
+    impl ObjectSubclass for FileSizePrivate {
+        const NAME: &'static str = "FileSize";
+        type ParentType = glib::Object;
+        type Instance = subclass::simple::InstanceStruct<Self>;
+        type Class = subclass::simple::ClassStruct<Self>;
+        type Type = super::FileSize;
+        glib::object_subclass!();
+
+        fn new() -> Self {
+            Self {
+                size: std::cell::RefCell::new(0),
+            }
+        }
+    }
+
+    impl ObjectImpl for FileSizePrivate {}
+
+    /// # Safety
+    ///
+    /// This is the ffi method to asynchronously get the file size. It accepts a callback of type
+    /// GAsyncReadyCallback, that will be invoked when the async operation completes. Typically,
+    /// this callback will invoke the get_file_size_finish method (implemented below) to get the
+    /// Task result and perform some operation with it.
+    #[no_mangle]
+    pub unsafe extern "C" fn get_file_size_async(
```

Comment:
I would move these into an `ffi` module and have them only do FFI: all actual functionality should be done via safe (public) Rust API that is outside the `imp` module.